src/Channels.cpp

```diff
@@ -228,14 +260,19 @@ PVR_ERROR Channels::GetChannelGroupMembers(const kodi::addon::PVRChannelGroup& g
   {
     tinyxml2::XMLNode* channelsNode = doc.RootElement()->FirstChildElement("channels");
     tinyxml2::XMLNode* pChannelNode;
-    for( pChannelNode = channelsNode->FirstChildElement("channel"); pChannelNode; pChannelNode=pChannelNode->NextSiblingElement())
+    for (pChannelNode = channelsNode->FirstChildElement("channel"); pChannelNode; pChannelNode = pChannelNode->NextSiblingElement())
     {
       kodi::addon::PVRChannelGroupMember tag;
-      tag.SetGroupName(group.GetGroupName());
       tag.SetChannelUniqueId(XMLUtils::GetUIntValue(pChannelNode, "id"));
-      tag.SetChannelNumber(XMLUtils::GetUIntValue(pChannelNode, "number"));
-      tag.SetSubChannelNumber(XMLUtils::GetUIntValue(pChannelNode, "minor"));
-      results.Add(tag);
+      // ignore orphan channels in groups
```

Comment:
Can you explain the comment? I'm not following the logic.

Reply:
The comment is because there is a bug in the backend and there could be channels returned that don't exist.  Currently these are logged as errors by core.  

m_channelDetails is existing code and it returns a tuple of  booleans for all channels, first is true when the channel has an EPG, second is true when it is a radio channel.  

The first test is existence

```
m_channelDetails.find(tag.GetChannelUniqueId()) != m_channelDetails.end() 
```

And the second test I want are true == true or false==false

```
group.GetIsRadio() == m_channelDetails[tag.GetChannelUniqueId()].second
```

The backend doesn't support radio or tv groups so I have to do it manually.  There is an assumption that the tv channels must be loaded before the tv group and the radio channel are loaded before radio groups.

Martin

Reply:
Ok, cool. Thanks for explaining.